crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1861,10 +1861,16 @@ fn handle_unary_op_comment<'a>(
             | SimpleTokenKind::Plus
             | SimpleTokenKind::Minus
     )));
-    let up_to = tokenizer
+    let lparen_start = tokenizer
         .find(|token| token.kind == SimpleTokenKind::LParen)
-        .map_or(unary_op.operand.start(), |lparen| lparen.start());
-    if comment.end() < up_to {
+        .map(|lparen| lparen.start());
+    let up_to = lparen_start.unwrap_or(unary_op.operand.start());
+    if lparen_start.is_none()
+        && comment.end() < unary_op.operand.start()
+        && comment.line_position().is_end_of_line()
+    {
+        CommentPlacement::dangling(unary_op, comment)
```

Comment:
I think we now need to update the documentation of this method and explain in which situation it makes comments as dangling